middleware/middleware.go

```diff
@@ -2,14 +2,21 @@ package middleware
 
 import (
 	"context"
+	"google.golang.org/grpc"
 )
 
 // Handler defines the handler invoked by Middleware.
 type Handler func(ctx context.Context, req interface{}) (interface{}, error)
 
+// StreamHandler defines the handler invoked by Middleware for stream calls.
+type StreamHandler func(srv interface{}, stream grpc.ServerStream) error
```

Comment:
Can I directly use the middleware interface? For example, putting the stream into req

Reply:
@wscalf has another proposal to unify the middleware interface: https://github.com/akoserwal/kratos/pull/1

Reply:
It's mostly scratchwork I wrote to check assumptions/make sure the interfaces lined up and in no way ready for review (I was holding off on spending a lot of time with it unless the Kratos team were interested in seeing more), but it goes along with what I was suggesting here: https://github.com/go-kratos/kratos/issues/3351#issuecomment-2223919198